src/test/java/org/example/DeliverToTest.java

```diff
@@ -0,0 +1,116 @@
+package org.example;
+
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+import java.util.List;
+
+import static org.testng.Assert.assertTrue;
+
+public class DeliverToTest {
+
+    public WebDriver webDriver;
+    public WebDriverWait wait;
+
+    //creating methods for setup/quit
+
+    @BeforeMethod
+    public void setup() {
+        System.setProperty(
+                "webdriver.chrome.driver",
+               System.getProperty("user.dir") + "\\src\\test\\resources\\WebDriver\\chromedriver.exe");
+
+        webDriver = new ChromeDriver();
+        webDriver.get("https://www.amazon.com/");
+        webDriver.manage().window().maximize();
+        wait = new WebDriverWait(webDriver, Duration.ofSeconds(10));
+    }
+
+    @AfterMethod
+    public void clean() {
+        webDriver.close();
+        webDriver.quit();
+    }
+
+    @Test
+    public void verifyZipCodeCorrectUpdateOnChangeInDeliverToSection() {
+        var zipCode = "36104";
+
+        WebElement deliverTo = webDriver.findElement(By.xpath("//div/span[@id= 'nav-global-location-data-modal-action']"));
+        deliverTo.click();
+
+        WebElement zipCodeInput = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("#GLUXZipUpdateInput")));
+
+        zipCodeInput.sendKeys(zipCode);
+
+        WebElement applyButton = webDriver.findElement(By.cssSelector("#GLUXZipUpdate"));
+        applyButton.click();
+
+        WebElement continueButton = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[contains(@class, 'a-popover-footer')]//input[contains(@id, 'GLUXConfirmClose')]")));
+
+        continueButton.click();
+
+        //waiting till the old element will be deleted from DOM, it indicates that the page was refreshed
+
+        boolean waitReload = wait.until(ExpectedConditions.stalenessOf(deliverTo));
+
+        WebElement textBox = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//span[@id= 'nav-global-location-data-modal-action']//span[contains(@id, 'glow-ingress-line2')]")));
+        Assert.assertTrue(textBox.getText().contains(zipCode));
```

Comment:
It is better to add a string that explains why the test failed. Same for other

Reply:
I don't understand, can you explain please? You mean to add a special message in case if test fails?

Reply:
WebElement textBox =
                wait.until(ExpectedConditions.visibilityOfElementLocated(
                        By.xpath("//span[@id= 'nav-global-location-data-modal-action']//span[contains(@id, 'glow-ingress-line2')]")
                ));
        
        Assert.assertTrue(textBox.getText().contains(zipCode), "Delivery To Textbox doesn't contain the required zipcode");


Looks better?

Reply:
Assert.assertTrue(textBox.getText().contains(zipCode), "Delivery To Textbox doesn't contain the required zipcode");

like this?